spotify/start.sh

```diff
@@ -4,4 +4,8 @@ if [[ -z "$BLUETOOTH_DEVICE_NAME" ]]; then
   BLUETOOTH_DEVICE_NAME=$(printf "balenaSound %s" $(hostname | cut -c -4))
 fi
 
-./usr/bin/librespot  --name "$BLUETOOTH_DEVICE_NAME" $BACKEND_ARGS --bitrate ${BITRATE} $CACHE_ARGS $VOLUME_ARGS
+# Set the system volume here
+SYSTEM_OUTPUT_VOLUME="${SYSTEM_OUTPUT_VOLUME:-100}"
+
+# Start raspotify
+./usr/bin/librespot  --name "$BLUETOOTH_DEVICE_NAME" --backend alsa --bitrate 320 --cache /var/cache/raspotify --enable-volume-normalisation --linear-volume --initial-volume=$SYSTEM_OUTPUT_VOLUME
```

Comment:
looks pretty good, I wonder if it would be a good idea to have the cache map to a docker volume, this can be done easily using `VOLUME /var/cache/raspotify` in the dockerfiles, but it might be even better to do it from the compose file as then the cache will persist between updates, etc.

Reply:
So should i add in the compose:

`
     volumes:
       - /var/cache/raspotify
`

Reply:
This would be fine with: `VOLUME /var/cache/raspotify` , but the compose even better, but i do not know how to implement something like this.

Reply:
Did you see this line in the Dockerfile? CMD /bin/mkdir -m 0755 -p /var/cache/raspotify ; /bin/chown raspotify:raspotify /var/cache/raspotify  It is from the official systemd service of raspotify which i implemented here. It runs on every boot and seems like overwriting the old cache. The cache should be different on every boot, so we do not need to change it.

Reply:
Ah, I didn't see that. Yeah in the case that its probably not useful to have the volume.

Reply:
Actually [that CMD line](https://github.com/balena-io-projects/balena-sound/blob/a993f570b5d0486d32aa72d0ab0e2ae6feed4f91/spotify/Dockerfile.template#L7) on the Dockerfile is not doing anything since it gets overriden by the CMD on the last line of the template. I think we should remove it to avoid confusion.

As for having the cache on a volume, I tested with a volume defined on the compose file and it works as intended. Not really sure it will make a big difference, but does not hurt either to add the option. If you don't provide a volume raspotify will create the cache folder nonetheless with the adecuate permissions, so no breaking changes there.  I'm all for adding it as an option.

@AlexProgrammerDE, you can check out [this](https://docs.docker.com/compose/compose-file/compose-file-v2/#volume-configuration-reference) reference to see how to add the cache as a volume. Just make sure you map the volume to /var/cache/raspotify. So you will have something that looks like this:

```
version: '2'

volumes:
  spotifycache:

services:
  ...
  spotify:
    build: ./spotify
    restart: always
    network_mode: host
    privileged: true
    labels:
      io.balena.features.dbus: 1
    volumes:
      - spotifycache:/var/cache/raspotify
```

Reply:
Also, [here](https://www.balena.io/docs/learn/develop/runtime/#balenaos-v2120-and-above) you can read about how balenaOS handles persistent storage through named volumes such as is this case.

Reply:
Looks good i will add your code. By the way how can i push new code without re triggering you CI with closing and reopening?

Reply:
CI will be triggered for each commit and that is fine. Not sure i follow what you mean with closing and reopening?

Reply:
Wait please i will show you it. I will now implement your code.

Reply:
I now tested and pushed it. In the past i had the problem that now your CI will not run the checks.

Reply:
I solved it with closing and reopening the PR.

Reply:
oh now i get it, you can retrigger it manually doing (at)balena-ci retest.
Thanks!

Reply:
Is there a way for me to retrigger it without closing and reopening?

Reply:
yes, you need to leave a comment on the PR conversation like this:

(at)balena-ci retest

Replace (at) with @, i don't want to trigger it :P

Reply:
OK i will try it.